zilliqa/src/exec.rs

```diff
@@ -1013,6 +1020,10 @@ impl PendingState {
         load_account(&self.pre_state, &mut self.new_state, address)
     }
 
+    pub fn network_id(&self) -> u64 {
+        self.pre_state.chain_id - 0x8000
```

Comment:
Can we move `zilliqa::api::zil::network_id` to a central place and use it both in the API layer and here? I don't want to have two sources of truth for the `- 0x8000` logic.